### DISCUSSION TITLE
Link Field improvements - add settings for a custom link text and a toggle to open in a new window, add the ability to set default link type

### DISCUSSION THREAD

**Discussion by nikolenko-dmitriy** [craftcms/cms#15686] [Ideas] [answered]
We tried to use native `Link Field` instead of `Typed Link Field` plugin and came to the conclusion that the logic with combining `Link Field`, plain text field `Custom link text` and lightswitch `Open in new window` into one matrix field is too complicated.

The reason is that there is too much to do if we need, for example, 5 link fields with their own unique settings.
For example, the list of allowed link types for them should be different.
In this case we can re-use only plain text field `Custom link text` and lightswitch `Open in new window`.

**But we will also need to do the following:**
- create 5 unique link fields
- create 5 unique matrix fields, that will relate to the appropriate link field
- create 5 unique entry types that will be used for the appropriate matrix field
- configure eveything in the control panel and re-use plain text field `Custom link text` and lightswitch `Open in new window`

It would be great if we could create just 5 unique link fields and not have to create matrix fields and additional entry types.

**There are 2 suggestions how to improve this:**

1. Add the ability for the native `Link Field` itself to have settings `Custom link text` and `Open in new window`
2. If the previous way doesn't work for you, would it be possible to add some new `UI Element` to the `Field Layout` to group fields into one box? I mean instead of adding a matrix field we could add `Link Field`, plain text field `Custom link text` and lightswitch `Open in new window` separately but visually it should be clear for the user they refer to the same thing. Below is the example of what we tried to do:
<img width="1110" alt="Screen Shot 2024-09-06 в 13 48 39" src="https://github.com/user-attachments/assets/d1c9c67f-d5b5-4454-8203-c5f75a90e759">

Another request is to add setting `Default link type` when configuring `Link Field`. Currently, the first option from the drop-down list is always selected by default.
<img width="450" alt="Screen Shot 2024-09-06 в 14 07 40" src="https://github.com/user-attachments/assets/4261d01a-5a14-48fe-83fc-f5b8e1ef02ab">

**Comment by brandonkelly** [comment]
Link attributes and labels were [intentionally omitted](https://github.com/craftcms/cms/pull/15400), but I’ll keep the discussion open in case we change out mind down the road.

The request to choose a default link type has been covered by #15516 and #15659.

**Reply by nikolenko-dmitriy** [reply]
I see, thank you 👍

**Comment by brandonkelly** [comment, marked as answer]
Added new “Show the ‘Label’ field” and “Show the ‘Open in a new tab’ field” settings to Link fields for Craft 5.5 (#15983) 🎉

**Reply by brandonkelly** [reply]
Craft 5.5.0 is out with those new Link field inputs 🎉

**Reply by adrienpicard** [reply]
Great addition thank you!

Could label+target be exposed via the GraphQL API too?

**Reply by brandonkelly** [reply]
Yep, already done for Craft 5.6 (#16237).
